test/unit_test/static/math_test.cpp

```diff
@@ -40,6 +40,13 @@ template<typename T1, typename T2, typename... Ts>
 
 #if __cpp_lib_constexpr_cmath || MP_UNITS_COMP_GCC
 
+static_assert(compare(fma(2 * m, 3 * m, 1 * m2), 7 * m2));
+static_assert(compare(fma(2.0 * s, 3.0 * Hz, 1.0 * one), 7.0));
+static_assert(compare(fma(2 * s, 3 * Hz, 1 * one), 7));
+static_assert(compare(fma(2.0*one, 3.0*m, 1.0*m), 7*m);
+static_assert(compare(fma(2.0*m, 3.0*one, 1.0*m), 7*m));
+// Works on Clang/AppleClang, but rejected by gcc:
+//static_assert(compare(fma(2.0*km, 1.0*km, 1e6*m2), 3.0*km));
```

Comment:
By the current implementation, the result should have the unit of the third argument.
With that done, Clang and GCC both accept: <https://cpp1.godbolt.org/z/9PEKfMrY4>.
If any is bound to be right going forward, it's GCC 13, as it implements <https://en.cppreference.com/w/cpp/header/stdfloat>.